### PR TITLE
NIAD-1603: AllergyIntolerance: Handle asserter as relatedPerson

### PR DESCRIPTION
**Pull request by bbell-kainos**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java [hunk 1]
```diff
@@ -193,6 +195,16 @@ private String buildAsserterPertinentInformation(AllergyIntolerance allergyIntol
             IIdType reference = allergyIntolerance.getAsserter().getReferenceElement();
             if (reference.getResourceType().equals(ResourceType.Patient.name())) {
                 return PATIENT_ASSERTER;
+            } else if (reference.getResourceType().equals(ResourceType.RelatedPerson.name())) {
+                RelatedPerson relatedPerson = messageContext
+                    .getInputBundleHolder()
+                    .getResource(reference)
+                    .map(RelatedPerson.class::cast)
+                    .get();
```
Comment (bartek-sarul) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java:203]: What if there is no such resource?
It should be something like this:
```
return messageContext
    .getInputBundleHolder()
    .getResource(reference)
    .map(RelatedPerson.class::cast)
    .filter(RelatedPerson::hasName)
    .map(RelatedPerson::getName)
    .map(names -> names.get(0))
    .map(HumanName::GetNameAsSingleString)
    .map(name -> String.format(RELATED_PERSON_ASSERTER, name))
    .orElse(StringUtils.EMPTY);
```

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java (hunk 1 above)
Comment (bbell-kainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java]: Addressed

**Review by bartek-sarul — APPROVED** [review]

**Review by ElliotB-96 — COMMENTED** [review]
Can you include a test where the relatedPerson does not have a name

**Review by bartek-sarul — APPROVED** [review]

**Comment by bbell-kainos** [comment]
> Can you include a test where the relatedPerson does not have a name

**Review by ElliotB-96 — APPROVED** [review]

**Pull request merged**